adapters/repos/db/lsmkv/store.go

```diff
@@ -616,3 +617,36 @@ func (s *Store) updateBucketDir(bucket *Bucket, bucketDir, newBucketDir string)
 		segment.setPath(updatePath(segment.getPath()))
 	}
 }
+
+// CalcTargetVectorDimensionsFromStore calculates dimensions and object count for a target vector from an LSMKV store
+func (s *Store) CalcTargetVectorDimensionsFromStore(ctx context.Context, targetVector string, calcEntry func(dimLen int, v []MapPair) (int, int)) (sum int, dimensions int) {
+	b := s.Bucket(helpers.DimensionsBucketLSM)
+	if b == nil {
+		return 0, 0
+	}
+
+	c := b.MapCursor()
+	defer c.Close()
+
+	var (
+		nameLen        = len(targetVector)
+		expectedKeyLen = 4 + nameLen
+	)
+
+	for k, v := c.First(ctx); k != nil; k, v = c.Next(ctx) {
+		// for named vectors we have to additionally check if the key is prefixed with the vector name
+		keyMatches := len(k) == expectedKeyLen && (nameLen == 4 || strings.HasPrefix(string(k), targetVector))
```

Comment:
could you elaborate on that check? why `nameLen == 4`?
EDIT: ok, i see method was moved